contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -2361,13 +2361,39 @@ object_id(PG_FUNCTION_ARGS)
 			} 
 			else if (enr == NULL)
 			{
-				result = get_relname_relid((const char *) object_name, LookupNamespaceNoError("pg_temp"));
+				Oid temp_nsp_oid = LookupNamespaceNoError("pg_temp");
+				if (OidIsValid(temp_nsp_oid))
+				{
+					result = get_relname_relid((const char *) object_name, temp_nsp_oid);
+				}
```

Comment:
Is this really necessary? I'm assuming even before if pg_temp lookup fails we would still set result as `InvalidOid`

Reply:
You're right it's not strictly needed, but I think it adds clarity and safety. Unless it slows things down, I'd like to keep it.